### PR TITLE
[#247] Refactor and integrate Anthropic Claude API

### PR DESCRIPTION
**Pull request by morcosan**
[#247] Refactor and add error handling for GPT API 
[#247] Implement creativity levels for Rammus and Lorem Ipsum GPT 
[#247] Add seed to random utilities
[#247] Update translations

### CODE REVIEW

File: src-api/services/gpt/_chatgpt-api.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { getDbAccount } from '@api/services/db'
+import { CreativityLevel, GptAPI, GptConfig, GptMessage, GptResponse } from '@api/types'
+import OpenAI, { APIError } from 'openai'
```
Comment (costinEEST) [inline src-api/services/gpt/_chatgpt-api.ts:3]:
```ts
import type { ChatModel as Model, ChatCompletionCreateParamsNonStreaming as Payload }  from 'openai'
```

File: src-api/services/db/_chats-db.ts [hunk 2]
```diff
@@ -145,9 +146,12 @@ const renameChat = async (chat: DbChat) => {
 			...messages.map((msg: DbMessage) => ({ text: msg.text, role: msg.role })),
 			{ text: 'Title this chat concisely in its spoken language, no quotes', role: 'user' },
 		]
-		const has4oMini = await ChatGPT4oMini.isAvailable()
+		let resp
 
-		if (has4oMini) return await ChatGPT4oMini.getResponse(config, gptMessages)
+		if (!resp && (await ChatGPT4oMini.isAvailable())) resp = await ChatGPT4oMini.getResponse(config, gptMessages)
+		if (!resp && (await Claude35Haiku.isAvailable())) resp = await Claude35Haiku.getResponse(config, gptMessages)
```
Comment (costinEEST) [inline src-api/services/db/_chats-db.ts:152]: `await` inside an `if` statement. It's the first time when I am seeing it, in a codebase.
I would ask an AI, if it is correct..

File: src-api/services/gpt/_claude-api.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import Anthropic, { APIError } from '@anthropic-ai/sdk'
+import Types from '@anthropic-ai/sdk/resources/messages'
+import { getDbAccount } from '@api/services/db'
+import { CreativityLevel, GptAPI, GptConfig, GptMessage, GptResponse } from '@api/types'
+
+type Message = Types.MessageParam
+type Model = Types.Model
+type Payload = Types.MessageCreateParamsNonStreaming
+type TextBlock = Types.TextBlock
```
Comment (costinEEST) [inline src-api/services/gpt/_claude-api.ts:9]: Again, just use import aliases.

**Review by costinEEST — APPROVED** [review]

File: src-api/services/db/_chats-db.ts (hunk 2 above)
Comment (morcosan) [reply src-api/services/db/_chats-db.ts:152]: it works :)))

File: src-api/services/gpt/_claude-api.ts (hunk 3 above)
Comment (morcosan) [reply src-api/services/gpt/_claude-api.ts:9]: I will refactor this in next PR

File: src-api/services/gpt/_chatgpt-api.ts (hunk 1 above)
Comment (morcosan) [reply src-api/services/gpt/_chatgpt-api.ts:3]: This one doesn't work. OpenAI is enforcing the namespace export.

File: src-api/services/db/_chats-db.ts (hunk 2 above)
Comment (morcosan) [reply src-api/services/db/_chats-db.ts:152]: But I will refactor than in next PR anyway.

**Pull request merged**